File: Algorithms/026_remove_duplicates_sorted_array.cc
class Solution {
public:
	int removeDuplicates(int A[], int n) {
		if(n == 0) return 0;
		int i(0), j(1), prev = INT_MAX;
		for(j = 0; j < n; ++j) {
			if(A[j] != prev) {
				A[i++] = A[j]; prev = A[j];
			}
		}
		return i;   
	}
};

File: Algorithms/java/026_remove_duplicates_sorted_array.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        //i: current element under investigation (jump over duplicates)
        //j: one past the end of processed array
        int n = nums.length;
        if (n <= 1) return n;
        int i = 0, j = 0;
        while (i < n) {
            while (i < n-1 && nums[i+1] == nums[i]) i++;
            nums[j++] = nums[i++];
        }
        return j;
    }
}
